sonic_platform_base/sonic_xcvr/api/public/c_cmis.py

```diff
@@ -0,0 +1,899 @@
+"""
+    c-cmis.py
+
+    Implementation of XcvrApi that corresponds to C-CMIS
+"""
+from ...fields import consts
+from .cmis import CmisApi
+import time
+BYTELENGTH = 8
+
+class CCmisApi(CmisApi):
+    def __init__(self, xcvr_eeprom):
+        super(CCmisApi, self).__init__(xcvr_eeprom)
+
+    def get_freq_grid(self):
+        '''
+        This function returns the configured frequency grid. Unit in GHz
+        '''
+        freq_grid = self.xcvr_eeprom.read(consts.GRID_SPACING)
```

Comment:
xcvr_eeprom.read() can also return None. If so, can you handle the conditions gracefully wherever read API is used.

Reply:
Done